[FEAT] App Dropdown 추가

devzoe opened the pull request:
## 📌 관련 이슈
<!-- 관련 이슈 번호를 입력해주세요. 예: #123 -->
#12 

## ✨ 작업 내용
<!-- 변경 사항을 간략히 요약해주세요.
   - 무엇을 변경했는지
   - 왜 변경했는지 -->
 - Flutter 기본 DropdownButton은 스타일 커스터마이징이나 드롭다운 위치 제어가 어려워 직접 구현했습니다.

- OverlayEntry로 드롭다운 렌더링
  - 기본 위젯은 부모 레이아웃 제약을 받아 드롭다운이 잘릴 수가 있어서 OverlayEntry를 활용해 드롭다운이 뜨도록 했습니다.
  - 위치는 RenderBox로 계산해서 클릭된 위젯 아래에 뜨게 처리했습니다.

- CompositedTransformTarget/Follower 사용
  - 드롭다운 위치가 기준 위젯에 자동으로 따라붙도록 CompositedTransform을 적용했습니다.

- Stateless 구조 + StatefulBuilder로 토글만 관리
  - 전체 컴포넌트는 StatelessWidget으로 구현하고, 드롭다운 열림 여부만 StatefulBuilder로 관리해서 불필요한 전체 리렌더링을 방지했습니다.

- dropdown width 가 2가지 버전이 있어서 호출하는 곳에서 넘겨주는 방식으로 구현했습니다. 상단 바는 dropdown width * 0.9 크기로 계산했습니다!

```dart
AppDropdown<String>(
    items: languages,
    value: selectedLanguage,
    onChanged: (val) {
        if (val != null) {
            setState(() {
                selectedLanguage = val;
            });
        }
    },
    itemBuilder: (item) => Text(item),
    icon: Assets.icon.info.earthLine,
    dropdownWidth: 200,
)
```

## 📸 스크린샷 (선택)

<!-- UI 변경 사항이 있다면 첨부해주세요. -->
| Dropdown(전국) - fold | Dropdown(전국) - unfold | Dropdown(서울) - unfold |
| ----------------------------------- | -------------------------------------- | --------------------- |
|  ![IMG_7279](https://github.com/user-attachments/assets/15c601df-1658-48d6-9910-1f9537bb25f4) |  ![IMG_7280](https://github.com/user-attachments/assets/1fc19945-262f-430e-9fde-fdbbbcc027b0) |  ![IMG_7281](https://github.com/user-attachments/assets/77c0f62a-d8cd-428a-89c6-981e86e3f2c6) |

| Dropdown(KR) - fold | Dropdown(KR) - unfold | Dropdown(EN) - unfold |
| ----------------------------------- | -------------------------------------- | --------------------- |
|  ![IMG_7282](https://github.com/user-attachments/assets/12771c02-bf2a-4c73-afdb-9637b4849186) |  ![IMG_7283](https://github.com/user-attachments/assets/3f0ee57b-e662-490c-bccf-7b14ad9e5cdf) |  ![IMG_7284](https://github.com/user-attachments/assets/de985616-6ac7-4456-8d7f-ba8702e87e98) |

---
| Dropdown(전국) - fold | Dropdown(숙명여대) - unfold | Dropdown(건국대) - unfold |
| ----------------------------------- | -------------------------------------- | --------------------- |
|  ![IMG_7323](https://github.com/user-attachments/assets/6bee8af2-ae77-4c57-91d2-0a3840da794f) |  ![IMG_7324](https://github.com/user-attachments/assets/4125ec00-370a-4751-a631-8fffb4ffa6f2)| ![IMG_7325](https://github.com/user-attachments/assets/4a82f80a-8cb1-46d3-b4c1-f94bb26977ea) |
## 📚 참고 자료
<!-- 참고한 가이드 문서나 레퍼런스 링크가 있다면 입력해주세요. -->

devzoe committed:
feat: App Dropdown

devzoe commented:
Remind Me(@devzoe ) : 드롭다운 아이템 - (학교 아이콘 - 학교 이름) -> 아이콘도 변경되도록 수정

generalChang commented:
너무 잘 구현해주신 것 같습니다! 😁
Dropdown의 외부 영역을 눌렀을때도 Dropdown List가 remove되면 좋을 것 같다는 생각입니다!

devzoe committed:
refactor: app dropdown 리팩토링
- 빈 화면 클릭하면 dropdownEntry remove하도록 GestureDetector 추가
- DropdownItem 추상 클래스 추가
- DropdownItem을 implements하는 UniversityItem 추가
- UniversityItem에서 svg, png 구분하여 각각 이미지 빌드하는 buildIcon 추가
